src/app/screens/gamepage/index.ts

```diff
@@ -75,17 +95,36 @@ export class GamePage {
     })
   }
 
-  private handleProgressBarChange() {
-    setInterval(() => {
-      const timeToNextWave = this.game.getNextAttack();
-      const width = timeToNextWave * 100 / 30;
-      this.progressBar.style.width = `${width}%`;
-      
-    }, 1000)
+  private startGame() {
+    this.game.start(this.updateUI.bind(this))
   }
 
-  private startGame() {
-    this.game.start()
-    this.handleProgressBarChange()
+  private updateUI() {
+    this.updateGoldAmount()
+    this.updateWoodAmount()
+    this.updateStoneAmount()
+    this.updatePopulation()
+    this.updateNextAttackProgressBar()
+  }
+
+  private updateGoldAmount() {
+    this.goldAmountElement.textContent = this.game.getGoldAmount().toString()
+  }
+
+  private updateWoodAmount() {
+    this.woodAmountElement.textContent = this.game.getWoodAmount().toString()
+  }
+
+  private updateStoneAmount() {
+    this.stoneAmountElement.textContent = this.game.getStoneAmount().toString()
+  }
+
+  private updatePopulation() {
+    this.populationElement.textContent = this.game.getPopulation().toString()
+  }
+
+  private updateNextAttackProgressBar() {
+    const width = (this.game.getNextAttack() * 100) / 30
```

Comment:
Maybe we can use here calculateNextAttack method from Game.ts

Reply:
what for? calculateNextAttack calculates only new countdown (in seconds) of next attack when current attack is finished.